### PR TITLE
feat(discourse): add users to group / change trust level / create post  / create topic / send private message actions

### PR DESCRIPTION
**Pull request by pfernandez98**
## What does this PR do?

This adds a new piece: Discourse.
For now, you can use the following actions:
* Add user(s) to groups
* Change trust level of user
* Send a PM to a user/list of users
* Send new Post
* Send new Topic

I did NOT add any triggers, because Discourse have a webhook feature, so you can use the AP webhooks to trigger ANY action.

ANNOUNCEMENT=true

### CODE REVIEW

File: packages/pieces/discourse/src/lib/actions/new-pm-to-user.action.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import { discourseAuth } from '../../index';
+import { Property, createAction } from "@activepieces/pieces-framework";
+
+
+
+export const newPrivateMessage = createAction({
+    auth: discourseAuth,
+    name: 'new_private_message',
+    description: 'Create a new private message in Discourse',
+    displayName: 'New Private Message',
```
Comment (abuaboud) [inline packages/pieces/discourse/src/lib/actions/new-pm-to-user.action.ts]: Can you please change to "Send Private Message"

File: packages/pieces/discourse/src/lib/actions/new-post.action.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+import { discourseAuth } from '../../index';
+import { Property, createAction } from "@activepieces/pieces-framework";
+
+
+
+export const newPost = createAction({
+    auth: discourseAuth,
+    name: 'new_post',
+    description: 'Create a new post in discourse',
+    displayName: 'New Post',
```
Comment (abuaboud) [inline packages/pieces/discourse/src/lib/actions/new-post.action.ts]: Can you please change to "Create Post", It's better to use verb with the action

File: packages/pieces/discourse/src/lib/actions/new-topic.action.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+import { discourseAuth } from '../../index';
+import { Property, createAction } from "@activepieces/pieces-framework";
+
+
+
+export const newTopic = createAction({
+    auth: discourseAuth,
+    name: 'new_topic',
+    description: 'Create a new topic in Discourse',
+    displayName: 'New Topic',
```
Comment (abuaboud) [inline packages/pieces/discourse/src/lib/actions/new-topic.action.ts]: Can you please change to "Create Topic", It's better to use verb with the action

File: packages/pieces/discourse/src/lib/actions/new-post.action.ts (hunk 2 above)
Comment (abuaboud) [reply packages/pieces/discourse/src/lib/actions/new-post.action.ts]: I would also do it for the files names and other actions as well :D

**Review by abuaboud — COMMENTED** [review]
Great Job! :D 

Left some minor comments let's do them and merge!

File: packages/pieces/discourse/src/lib/actions/new-pm-to-user.action.ts (hunk 1 above)
Comment (pfernandez98) [reply packages/pieces/discourse/src/lib/actions/new-pm-to-user.action.ts]: done

File: packages/pieces/discourse/src/lib/actions/new-post.action.ts (hunk 2 above)
Comment (pfernandez98) [reply packages/pieces/discourse/src/lib/actions/new-post.action.ts]: done

File: packages/pieces/discourse/src/lib/actions/new-topic.action.ts (hunk 3 above)
Comment (pfernandez98) [reply packages/pieces/discourse/src/lib/actions/new-topic.action.ts]: done

**Review by abuaboud — APPROVED** [review]

**Pull request merged**